File: algo/src/main/java/com/hartwig/actin/algo/evaluation/general/WillParticipateInTrialInCountry.java
package com.hartwig.actin.algo.evaluation.general;

import com.hartwig.actin.PatientRecord;
import com.hartwig.actin.algo.datamodel.Evaluation;
import com.hartwig.actin.algo.datamodel.EvaluationResult;
import com.hartwig.actin.algo.datamodel.ImmutableEvaluation;
import com.hartwig.actin.algo.evaluation.EvaluationFunction;

import org.jetbrains.annotations.NotNull;

public class WillParticipateInTrialInCountry implements EvaluationFunction {

    @NotNull
    private final String country;

    WillParticipateInTrialInCountry(@NotNull final String country) {
        this.country = country;
    }

    @NotNull
    @Override
    public Evaluation evaluate(@NotNull PatientRecord record) {
        EvaluationResult result = country.toLowerCase().contains("netherlands") ? EvaluationResult.PASS : EvaluationResult.FAIL;

        ImmutableEvaluation.Builder builder = ImmutableEvaluation.builder().result(result);
        if (result == EvaluationResult.FAIL) {
            builder.addFailSpecificMessages("Patient will not be participating in " + country);
        } else if (result == EvaluationResult.PASS) {
            builder.addPassSpecificMessages("Patient will be participating in " + country);
        }

        return builder.build();
    }
}

File: algo/src/main/kotlin/com/hartwig/actin/algo/evaluation/general/WillParticipateInTrialInCountry.kt
package com.hartwig.actin.algo.evaluation.general

import com.hartwig.actin.algo.evaluation.EvaluationFactory
import com.hartwig.actin.algo.evaluation.EvaluationFunction
import com.hartwig.actin.algo.evaluation.EvaluationLabels
import com.hartwig.actin.datamodel.PatientRecord
import com.hartwig.actin.datamodel.algo.Evaluation

class WillParticipateInTrialInCountry(private val country: String, private val labels: EvaluationLabels.General) : EvaluationFunction {

    override fun evaluate(record: PatientRecord): Evaluation {
        return when {
            country.lowercase().contains("netherlands") -> EvaluationFactory.pass(labels.willParticipateInTrialInCountryPass(country))

            else -> EvaluationFactory.fail(labels.willParticipateInTrialInCountryFail(country))
        }
    }
}